ssclj/jar/src/main/clojure/com/sixsq/slipstream/ssclj/middleware/proxy_redirect.clj

```diff
@@ -41,9 +42,23 @@
       (.read rdr buf)
       buf)))
 
-(defn- slurp-body-binary
+(defn- not-delete?
   [request]
-  (if-let [len (when (:body request) (get-in request [:headers "content-length"]))]
+  (not= :delete (:request-method request)))
+
+(defn- body?
+  [request]
+  (:body request))
```

Comment:
This `body?` is not a predicate in the canonical sense.  Maybe better make it to return boolean.  Something like
```
(not (nil? (:body request)))
```